src/client/Microsoft.Identity.Client/Kerberos/KerberosSupplementalTicketManager.cs

```diff
@@ -0,0 +1,220 @@
+        /// <returns>Byte stream representaion of KRB-CRED Kerberos Ticket if it contains valid ticket information.
+        /// Null, otherwise.</returns>
+        public static byte[] GetKrbCred(KerberosSupplementalTicket ticket)
+        {
+            if (!string.IsNullOrEmpty(ticket.KerberosMessageBuffer))
+            {
+                return Convert.FromBase64String(ticket.KerberosMessageBuffer);
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Add Claims to body parameter for POST request.
+        /// </summary>
+        /// <param name="oAuth2Client"><see cref="OAuth2Client"/> object for Token request.</param>
+        /// <param name="requestParams"><see cref="AuthenticationRequestParameters"/> containing request parameters.</param>
+        internal static void AddKerberosTicketClaim(
+            OAuth2Client oAuth2Client,
+            AuthenticationRequestParameters requestParams)
+        {
+            string kerberosClaim = GetKerberosTicketClaim(
+                requestParams.RequestContext.ServiceBundle.Config.KerberosServicePrincipalName,
+                requestParams.RequestContext.ServiceBundle.Config.TicketContainer);
+
+            if (string.IsNullOrEmpty(kerberosClaim))
+            {
+                oAuth2Client.AddBodyParameter(OAuth2Parameter.Claims, requestParams.ClaimsAndClientCapabilities);
+            }
+            else if (string.IsNullOrEmpty(requestParams.ClaimsAndClientCapabilities))
+            {
+                oAuth2Client.AddBodyParameter(OAuth2Parameter.Claims, kerberosClaim);
+            }
+            else
+            {
+                JObject existingClaims = JObject.Parse(requestParams.ClaimsAndClientCapabilities);
+                JObject mergedClaims
+                    = ClaimsHelper.MergeClaimsIntoCapabilityJson(kerberosClaim, existingClaims);
+
+                oAuth2Client.AddBodyParameter(OAuth2Parameter.Claims, mergedClaims.ToString(Formatting.None));
+            }
+        }
+
+        /// <summary>
+        /// Generate a Kerberos Ticket Claim string.
+        /// </summary>
+        /// <param name="servicePrincipalName">Service principal name to use.</param>
+        /// <param name="ticketContainer">Ticket container to use.</param>
+        /// <returns>A Kerberos Ticket Claim string if valid service principal name was given. Empty string, otherwise.</returns>
+        internal static string GetKerberosTicketClaim(string servicePrincipalName, KerberosTicketContainer ticketContainer)
+        {
+            if (string.IsNullOrEmpty(servicePrincipalName))
+            {
+                return string.Empty;
+            }
+
+            if (ticketContainer == KerberosTicketContainer.IdToken)
+            {
+                return string.Format(
+                    CultureInfo.InvariantCulture,
+                    IdTokenAsRepTemplate,
+                    servicePrincipalName);
+            }
+            else

```

Comment:
Since you are returning in the if statement above, you probably dont need the else.

Reply:
Updated.